exercises/grains/src/example.c

```diff
@@ -0,0 +1,18 @@
+#include "grains.h"
+
+#define NUMBER_OF_SQUARES   (64)
+
+unsigned long square(int index) {
+  return 1ul << (index - 1);
+}
+
+unsigned long total(void) {
+  unsigned long result = 0;
+  int i;
+
+  for(i = 1; i <= NUMBER_OF_SQUARES; i++) {
```

Comment:
We should also decide what C standard we're targeting. I'm partial to C99, but I'll create an issue to discuss.